File: trunk/Client/Assets/Script/Panels/HeroItem.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using Assets.Script.Frame;
using UnityEngine.UI;

public class HeroItem : PanelBase
{
    
    Image ImageHeroHead;
    Image ImageHeroQuality;
    Image ImageHeroOccupation;
    Text TextHeroName;
    Text TextHeroLevel;
    Text TextChips;
    Text HeroLevel;
    Slider SliderChips;



    public void GetControls()
    {
        ImageHeroHead = GetImage("ImageHeroHead");
        ImageHeroQuality = GetImage("ImageHeroQuality");
        ImageHeroOccupation = GetImage("ImageHeroOccupation");
        HeroLevel = GetText("HeroLevel");
        TextHeroName = GetText("TextHeroName");
        TextHeroLevel = GetText("TextHeroLevel");
        TextChips = GetText("TextChips");
        SliderChips = controls.GetSlider("SliderChips");


        //txCoin = GetText("TextCoin");
        //txDiamond = GetText("TextDiamond");
        //txVigour = GetText("TextVigour");
    }

    public void SetHeroInfo(HeroInfo heroInfo)
    {
        GetControls();
        int heroResData = heroInfo.Id & 0xffff;
        HeroTableData heroTableData = Database.me.GetById<HeroTableData>(heroResData);
        ImageHeroHead.sprite = Assets.Script.SpriteFactory.me.Get("HeroHead", heroResData);
        ImageHeroQuality.sprite = Assets.Script.SpriteFactory.me.Get("HeroQuality", heroInfo.Quality);
        ImageHeroOccupation.sprite = Assets.Script.SpriteFactory.me.Get("HeroOccupation", heroTableData.HeroOccupation);
        TextHeroName.text = heroTableData.HeroName;
        TextHeroLevel.text = "Lv." + heroInfo.Level.ToString();
        HeroLevel.gameObject.SetActive(true);
        SliderChips.gameObject.SetActive(false);
    }

    public void SetHeroData(HeroTableData heroData)
    {
        GetControls();
        HeroLevel.gameObject.SetActive(false);
        SliderChips.gameObject.SetActive(true);
        ImageHeroHead.sprite = Assets.Script.SpriteFactory.me.Get("HeroHead", heroData.Id);
        ImageHeroQuality.sprite = Assets.Script.SpriteFactory.me.Get("HeroQuality", 1);
        ImageHeroOccupation.sprite = Assets.Script.SpriteFactory.me.Get("HeroOccupation", heroData.HeroOccupation);
        TextHeroName.text = heroData.HeroName;
    }

 
}

File: trunk/Client/Lua/HeroItem.lua
-- HeroItem.lua
print("HeroItem.lua loaded")


local GameObject = UnityEngine.GameObject
local LuaPanel = Assets.Script.Frame.LuaPanel
local panelMgr = Assets.Script.Frame.PanelManager.me
local database = Database.me;
local gamedata = Gamedata.me;
local spriteFactory = Assets.Script.SpriteFactory.me;

HeroItem = {}
HeroItem.__index = HeroItem;

function HeroItem.New(base)
    print("HeroItem.New")
    o = {}
    o.base = base;
    setmetatable(o,HeroItem);
    print(o.Init);
    return o;
end

function HeroItem:Init()
	Debugger.Log("HeroItem.Init")
	Debugger.Log(self.base.controls);
	local controls = self.base.controls;

	self.ImageHeroHead = controls:GetImage("ImageHeroHead");
    self.ImageHeroQuality = controls:GetImage("ImageHeroQuality");
    self.ImageHeroOccupation = controls:GetImage("ImageHeroOccupation");
    self.HeroLevel = controls:GetText("HeroLevel");
    self.TextHeroName = controls:GetText("TextHeroName");
    self.TextHeroLevel = controls:GetText("TextHeroLevel");
    self.TextChips = controls:GetText("TextChips");
    self.SliderChips = controls:GetSlider("SliderChips");

end



function HeroItem:SetHeroInfo(heroInfo)
    Debugger.Log("HeroItem.SetHeroInfo")
    local heroResData = heroInfo.Id % 0x10000;
    local heroTableData = database:GetById(typeof(HeroTableData), heroResData);
    self.ImageHeroHead.sprite = spriteFactory:Get("HeroHead", heroResData);
    self.ImageHeroQuality.sprite = spriteFactory:Get("HeroQuality", heroInfo.Quality);
    self.ImageHeroOccupation.sprite = spriteFactory:Get("HeroOccupation", heroTableData.HeroOccupation);
    self.TextHeroName.text = heroTableData.HeroName;
    self.TextHeroLevel.text = "Lv."..tostring(heroInfo.Level);
    self.HeroLevel.gameObject:SetActive(true);
    self.SliderChips.gameObject:SetActive(false);
end


function HeroItem:SetHeroData(heroData)
 	Debugger.Log("HeroItem.SetHeroData")
    self.ImageHeroHead.sprite = spriteFactory:Get("HeroHead", heroData.Id);
    self.ImageHeroQuality.sprite = spriteFactory:Get("HeroQuality", 1);
    self.ImageHeroOccupation.sprite = spriteFactory:Get("HeroOccupation", heroData.HeroOccupation);
    self.TextHeroName.text = heroData.HeroName;
    self.HeroLevel.gameObject:SetActive(false);
    self.SliderChips.gameObject:SetActive(true);
end
